common/app/com/commercetools/sunrise/framework/viewmodels/ViewModelFactory.java

```diff
@@ -1,5 +1,49 @@
 package com.commercetools.sunrise.framework.viewmodels;
 
+import com.commercetools.sunrise.framework.viewmodels.forms.MessageType;
+import com.commercetools.sunrise.framework.viewmodels.forms.MessageViewModel;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static play.mvc.Http.Context.Implicit.flash;
+
 public abstract class ViewModelFactory {
 
+    /**
+     * Extracts all messages saved via
+     * {@link com.commercetools.sunrise.framework.controllers.WithFormFlow#saveMessage(MessageType, String)}
+     * and returns a list of view models with those messages. These view models can be set for example in the
+     * {@link com.commercetools.sunrise.framework.viewmodels.content.PageContent} using
+     * {@link com.commercetools.sunrise.framework.viewmodels.content.PageContent#setMessages(List)}
+     *
+     * @return list of view models with the extracted messages
+     */
+    protected static List<MessageViewModel> extractMessages() {
+        return extractMessages(flash());
+    }
+
+    private static List<MessageViewModel> extractMessages(final Map<String, String> map) {
```

Comment:
We can have this method protected, but is it really useful? The point of the one above is that it is part of the mechanism of saving/displaying messages, but a method that extracts messages from a map, I'm not sure where can this be applied :\
WDYT?

Reply:
just to leave note, I think keeping it here is fine until the need for refactoring rise if this method proves useful 👍